R/parse_system.R

```diff
@@ -79,14 +79,14 @@ parse_system_overall <- function(exprs, call) {
   }
 
   dims <- lapply(exprs[is_dim], function(x) x$rhs$value)
-  arrays <- data_frame(
+  arrays <- resolve_array_references(data_frame(
```

Comment:
One of the big advantages of allowing

```
dim(a) <- dim(b)
```

is to stop the proliferation of data that hold the same value.  So perhaps here add a column "alias" or something that says that `b` uses the dimensions of `a`, or a column saying that `b` is not real so that we don't save it (that second option might be better)

Reply:
Another thing to consider is how this interacts with the dependencies, particularly in the case where we have

```
dim(a) <- parameter(rank = 2)
dim(b) <- dim(a)
```

We will need to process `a` before `b`